yaml_edit/lib/src/editor.dart

```diff
@@ -417,11 +417,11 @@ class YamlEditor {
   /// - 2 # comment 2
   /// '''
   /// ```
-  YamlNode remove(Iterable<Object> path) {
+  YamlNode? remove(Iterable<Object?> path) {
```

Comment:
I also began looking at public methods where `YamlNode?` are returned as well. For `remove()`, the only documentation I can see about the `null` path is on the comment at line 430:

```dart
/// Parsing an empty YAML document returns `null`.
_performEdit(edit, path, expectedNode);
```

It also doesn't have tests that demonstrate using the return value of `remove()`. :(

Reply:
BTW, this was the only public method on editor that still returns `YamlNode?`. Everything else either is just `void` or something else and `Iterable<YamlNode>`.

Reply:
So when will this return `null`? If the document is empty and the `path` is empty, then it doesn't return `null` does it?

It returns the `YamlNode` that has been removed. So unless the path is invalid there is an `YamlNode` that the path points to, and that node should be removed and returned.

Reply:
The signature has been changed and this no longer applies. It now returns just `YamlNode`.